src/controllers/post.controller.ts

```diff
@@ -16,6 +16,17 @@ class PostController {
       next(error);
     }
   };
-}
 
-export default PostController;
+  public updatePost = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+    try {
+      const userId = req.user.id
+      const updatedPostData: BasePostDto = req.body;
+      const update: BasePostDto = await this.postService.updatePost({...updatedPostData, userId}, userId);
+      res.status(201).json({ update, message: 'success'})
```

Comment:
Can we update the variable to data and have `res.status(201).json({ data, message: 'success'})`
Thinking this would be easier to work with in the frontend